### PR TITLE
Drop down menu

### PR DESCRIPTION
**Pull request by greglemonmapbox**
**Related issue**

**Description of changes**

New dropdown menu UI for Abstract Map modifier selection.

**QA checklists**

- [x] Add relevant code comments. Every API class and method should have `<summary>` description as well as description of parameters.
- [ ] **Add tests for new/changed/updated classes and methods!!!**
- [x] Check out conventions in [CONTRIBUTING.md](https://github.com/mapbox/mapbox-unity-sdk/blob/develop/CONTRIBUTING.md).
- [x] Check out conventions in [CODING-STYLE.md](https://github.com/mapbox/mapbox-unity-sdk/blob/develop/CODING-STYLE.md)
- [x] Update the [changelog](https://github.com/mapbox/mapbox-unity-sdk/blob/develop/documentation/docs/05-changelog.md)
- [ ] Update documentation.

**Reviewers**

@atripathi-mb

### CODE REVIEW

File: sdkproject/Assets/Mapbox/Unity/Editor/CreateClass.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
+﻿using UnityEditor;
```
Comment (atripathi-mb) [inline sdkproject/Assets/Mapbox/Unity/Editor/CreateClass.cs]: Add namespace.

File: sdkproject/Assets/Mapbox/Unity/Editor/CreateClass.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+using UnityEngine;
+using System.Collections;
+using System;
+using System.IO;
+using System.Text;
+
+public static class CreateClass
```
Comment (atripathi-mb) [inline sdkproject/Assets/Mapbox/Unity/Editor/CreateClass.cs]: Class name too generic. Please change to something more descriptive.

File: sdkproject/Assets/Mapbox/Unity/Editor/EditorWindowWithPopup.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
+﻿using UnityEngine;
```
Comment (atripathi-mb) [inline sdkproject/Assets/Mapbox/Unity/Editor/EditorWindowWithPopup.cs]: Is this required? Looks like leftover code.

File: sdkproject/Assets/Mapbox/Unity/Editor/PopupSelectionMenu.cs [hunk 4]
```diff
@@ -0,0 +1,130 @@
+﻿using UnityEngine;
```
Comment (atripathi-mb) [inline sdkproject/Assets/Mapbox/Unity/Editor/PopupSelectionMenu.cs]: Add namespace.

File: sdkproject/Assets/Mapbox/Unity/Editor/PopupSelectionMenu.cs [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+	public override Vector2 GetWindowSize()
+	{
+		return new Vector2(250, 250);
+	}
+
+	public override void OnGUI(Rect rect)
+	{
+		if (_modTypes == null || _modTypes.Count == 0)
+		{
+			var list = AssetDatabase.FindAssets("t:" + _type.Name);
```
Comment (atripathi-mb) [inline sdkproject/Assets/Mapbox/Unity/Editor/PopupSelectionMenu.cs]: According to this logic, you are only displaying Modifiers which exist in the project as scriptable objects. 
Fails when no `Modifier` scriptable objects in scene - the dropdown will be empty. This menu should always show all the possible modifiers available in the SDK + any custom modifiers created by user. 

Using reflection to find all modifier classes is one possible solution.

File: sdkproject/Assets/Mapbox/Unity/Editor/PopupSelectionMenu.cs [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+				var ne = AssetDatabase.GUIDToAssetPath(item);
+				var asset = AssetDatabase.LoadAssetAtPath(ne, _type) as ScriptableObject;
+				if(!_modTypes.Contains(asset.GetType()))
+				{
+					_modTypes.Add(asset.GetType());
+				}
+			}
+		}
+
+		else
```
Comment (atripathi-mb) [inline sdkproject/Assets/Mapbox/Unity/Editor/PopupSelectionMenu.cs]: Unnecessary else? removing this else fixes the repaint error

File: sdkproject/ProjectSettings/ProjectVersion.txt [hunk 7]
```diff
@@ -1 +1 @@
-m_EditorVersion: 2017.3.1f1
+m_EditorVersion: 2017.4.0f1
```
Comment (atripathi-mb) [inline sdkproject/ProjectSettings/ProjectVersion.txt]: Don't check in projectVersion file please.

File: sdkproject/Assets/Mapbox/Unity/Editor/PopupSelectionMenu.cs (hunk 4 above)
Comment (greglemonmapbox) [reply sdkproject/Assets/Mapbox/Unity/Editor/PopupSelectionMenu.cs]: namespace Mapbox.Editor added.

**Pull request merged**